storage/files.js

```diff
@@ -72,326 +68,281 @@ function listFiles (name, callback) {
+var cli = require('yargs');
+
+var program = module.exports = {
   listFiles: listFiles,
-  listFilesWithPrefix: listFilesWithPrefix,
+  listFilesByPrefix: listFilesByPrefix,
   uploadFile: uploadFile,
   downloadFile: downloadFile,
   deleteFile: deleteFile,
   getMetadata: getMetadata,
   makePublic: makePublic,
   moveFile: moveFile,
   copyFile: copyFile,
-  printUsage: printUsage,
-
-  // Executed when this program is run from the command-line
-  main: function (args, cb) {
-    var command = args.shift();
-    if (command === 'list') {
-      this.listFiles(args[0], cb);
-    } else if (command === 'listByPrefix') {
-      this.listFilesWithPrefix(args[0], args[1], args[2], cb);
-    } else if (command === 'upload') {
-      this.uploadFile(args[0], args[1], cb);
-    } else if (command === 'download') {
-      this.downloadFile(args[0], args[1], args[2], cb);
-    } else if (command === 'delete') {
-      this.deleteFile(args[0], args[1], cb);
-    } else if (command === 'getMetadata') {
-      this.getMetadata(args[0], args[1], cb);
-    } else if (command === 'makePublic') {
-      this.makePublic(args[0], args[1], cb);
-    } else if (command === 'move') {
-      this.moveFile(args[0], args[1], args[2], cb);
-    } else if (command === 'copy') {
-      this.copyFile(args[0], args[1], args[2], args[3], cb);
-    } else {
-      this.printUsage();
-    }
+  main: function (args) {
+    // Run the command-line program
+    cli.help().strict().parse(args).argv;
   }
 };
 
+cli
+  .demand(1)
+  .command('list <bucket> [options]', 'List files in a bucket, optionally filtering by a prefix.', {
+    prefix: {
+      alias: 'p',
+      requiresArg: true,
+      type: 'string',
+      description: 'Filter files by a prefix.'
+    },
+    delimiter: {
+      alias: 'd',
+      requiresArg: true,
+      type: 'string',
+      description: 'Specify a delimiter.'
+    }
+  }, function (options) {
+    if (options.prefix !== undefined) {
+      program.listFilesByPrefix(options, console.log);
+    } else {
+      program.listFiles(options.bucket, console.log);
+    }
+  })
+  .command('upload <bucket> <srcFile>', 'Upload a local file to a bucket.', {}, function (options) {
+    program.uploadFile(options, console.log);
+  })
+  .command('download <bucket> <srcFile> <destFile>', 'Download a file from a bucket.', {}, function (options) {
+    program.downloadFile(options, console.log);
+  })
+  .command('delete <bucket> <file>', 'Delete a file from a bucket.', {}, function (options) {
+    program.deleteFile(options, console.log);
+  })
+  .command('getMetadata <bucket> <file>', 'Get metadata for a file in a bucket.', {}, function (options) {
+    program.getMetadata(options, console.log);
+  })
+  .command('makePublic <bucket> <file>', 'Make a file public in a bucket.', {}, function (options) {
+    program.makePublic(options, console.log);
+  })
+  .command('move <bucket> <srcFile> <destFile>', 'Rename a file in a bucket.', {}, function (options) {
```

Comment:
The Cloud Storage API (and `gsutil` CLI) mirror the S3 API and the also the linux `mv` command somewhat, e.g. `mv old_file.txt new_file.txt`. I think `move` is fine here. I'll clarify in the description though.